main.cpp

```diff
@@ -53,20 +53,278 @@ Project 4: Part 4 / 9
-    Point& multiply(float m)
+    explicit IntType(int v) : value(new int(v)) {}
+    ~IntType()
     {
-        x *= m;
-        y *= m;
-        return *this;
+        delete value;
     }
+
+    IntType& add( int rhs );
+    IntType& subtract( int rhs );
+    IntType& multiply( int rhs );
+    IntType& divide( int rhs );
+    operator int() const { return *value; }
+    
+    IntType& pow(const FloatType&);
+    IntType& pow(const DoubleType&);
+    IntType& pow(const IntType&);
+    IntType& pow(int i);
+
+private:
+    IntType& pow_internal(const int exp);
+    int* value = nullptr;
+};
+
+DoubleType& DoubleType::add(double rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+DoubleType& DoubleType::subtract(double rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+DoubleType& DoubleType::multiply(double rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+DoubleType& DoubleType::divide(double rhs) 
+{ 
+    if (rhs == 0.0)
+        { std::cout << "warning: floating point division by zero!" << std::endl; }
+    *value /= rhs;
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const FloatType& ft)
+{
+    return pow_internal(static_cast<double>(ft));
+}
+
+DoubleType& DoubleType::pow(const DoubleType& dt)
+{
+    return pow_internal(dt);
+}
+
+DoubleType& DoubleType::pow(const IntType& it)
+{
+    return pow_internal(static_cast<double>(it));
+}
+
+DoubleType& DoubleType::pow(double dbl)
+{
+    return pow_internal(dbl);
+}
+
+DoubleType& DoubleType::pow_internal(const double exp)
+{
+    *value = std::pow(*value, exp);
+    return *this;
+}
+
+FloatType& FloatType::add(float rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+
+FloatType& FloatType::subtract(float rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+
+FloatType& FloatType::multiply(float rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+
+FloatType& FloatType::divide(float rhs) 
+{ 
+    if (rhs == 0.f) 
+        { std::cout << "warning: floating point division by zero!" << std::endl; }
+    *value /= rhs;
+    return *this;
+}
+
+FloatType& FloatType::pow(const FloatType& ft)
+{
+    return pow_internal(ft);
+}
+
+FloatType& FloatType::pow(const DoubleType& dt)
+{
+    return pow_internal(static_cast<float>(dt));
+}
+
+FloatType& FloatType::pow(const IntType& it)
+{
+    return pow_internal(static_cast<float>(it));
+}
+
+FloatType& FloatType::pow(float flt)
+{
+    return pow_internal(flt);
+}
+
+FloatType& FloatType::pow_internal(const float exp)
+{
+    *value = std::pow(*value, exp);
+    return *this;
+}
+
+
+
+IntType& IntType::add(int rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+IntType& IntType::subtract(int rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+IntType& IntType::multiply(int rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+IntType& IntType::divide(int rhs) 
+{ 
+    if (rhs == 0)
+        { std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl; }
+    else 
+        { *value /= rhs; }
```

Comment:
Fix this coding style
Conform to the course coding standard
Curly braces go on their online with no other code. 
- [ ] revice